src/zope/security/tests/test_proxy.py

```diff
@@ -2007,7 +2014,7 @@ def test_binops(self):
                     else:
                         self.assertEqual(removeSecurityProxy(eval(expr)), z,
                                          "x=%r; y=%r; expr=%r" % (x, y, expr))
-                        self.shouldFail(lambda x, y: eval(expr), x, y)
+                        self.shouldFail(self._make_eval(expr, locals()), x, y)
```

Comment:
I don't understand the purpose of these changes.

Reply:
Also a linter error: ["Cell variable 'expr' defined in loop"](https://stackoverflow.com/questions/25314547/cell-var-from-loop-warning-from-pylint).

Here, because the lambda is used immediately and then discarded, it doesn't cause any problems. But if that should change for any reason, the results can be very confusing (i.e., it would only actually test one `expr`). This has bitten me before and it jumped out at me here.

Reply:
Oh, I see!  I didn't notice the loop.

Wouldn't it have been simpler to do the traditional `lambda x, y, expr=expr: eval(expr)`? workaround instead of introducing a complicated helper maker and passing `locals()`?

Reply:
Yes. Yes it would have been. I have no idea why I wrote it that way.

Reply:
Ok, all I wanted to know if I was missing something ;)